java/test/jmri/util/WaitHandlerTest.java

```diff
@@ -23,15 +23,16 @@ public class WaitHandlerTest {
 
     static final int THREAD_DELAY = 500;   // time to delay thread under test
 
-    transient long startTime;
-    transient long endTime;
+    private transient long startTime;
+    private transient long endTime;
 
     @Test
     public void testInlineWait() {
         startTime = Calendar.getInstance().getTimeInMillis();
 
         // delay the test thread itself
-        new WaitHandler(this, 50);
+        WaitHandler t = new WaitHandler(this, 50);
+        Assertions.assertNotNull(t);
```

Comment:
Can a Java `new` ever return null?

Reply:
> Can a Java `new` ever return null?

No. If I remember correctly, it was @pabender who argumented about this practice. For some reason, he wanted an assert in every test method.

One reason might be compiler optimization. By having the assert method, you know that the compiler must create the object since it's used in the call to the assert method.

Reply:
I've commented in PR, not this thread whoops.

Reply:
> > Can a Java `new` ever return null?
> 
> No. If I remember correctly, it was @pabender who argumented about this practice. For some reason, he wanted an assert in every test method.
> 

It’s not just me. 

A test that can’t fail isn’t generally considered a valid test.  You can’t fail if you don’t have an assertion.

( for someone else who wrote on this see: 
https://developer.guidewire.com/best-practices-for-assertions-in-unit-tests/ 

there are tons of other people who have written about this.  )

Generally when you are using test driven development (TDD), the first step is to make a new test that fails.  You then write the least amount of code to make the test pass, then you repeat the process ( by creating either a new test or another assertion that fails)

Reply:
> A test that can’t fail isn’t generally considered a valid test. You can’t fail if you don’t have an assertion.

That's plain wrong.

```
package jmri.jmrit.logixng.actions.configurexml;

import org.junit.Test;

public class CtorTest {

    public static class MyClass {
        public MyClass() {
            throw new RuntimeException("Daniel");
        }
    }

    @Test
    public void testCtor() {
        new MyClass();
    }

}
```

This test fails with:
```
Errors: 
  CtorTest.testCtor:15 » Runtime Daniel
```